12/umbraco-cms/reference/api-versioning-and-openapi.md

```diff
@@ -0,0 +1,346 @@
+    }
+}
+
+public static class MySchemaIdUmbracoBuilderExtensions
+{
+    public static IUmbracoBuilder ConfigureMySchemaId(this IUmbracoBuilder builder)
+    {
+        // call this from ConfigureServices() in Startup, i.e.:
+        //     services.AddUmbraco(_env, _config)
+        //         ...
+        //         .ConfigureMySchemaId()
+        //         .Build();
+        builder.Services.AddSingleton<ISchemaIdSelector, MySchemaIdSelector>();
+        return builder;
+    }
+}
+```
+{% endcode %}
+
+## Adding your own Swagger documents
+
+Umbraco automatically adds a "default" Swagger document to contain all APIs that are not explicitly mapped to a named Swagger document. This means that your custom APIs will automatically appear in the "default" Swagger document.
+
+If you want to exercise more control over where your APIs show up in Swagger, you can do so by adding your own Swagger documents.
+
+{% hint style="info" %}
+Umbraco imposes no limitations on adding Swagger documents, and the code below is a simplistic example.
+
+In the [Swashbuckle GitHib repository](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/) you will find comprehensive documentation for Swagger documents.
+{% endhint %}
+
+A common use case for this is when you maintain multiple versions of the same API. Often you want to have separate Swagger documents for each version. The following code sample creates two Swagger documents - "My API v1" and "My API v2".
+
+{% code title="How to create your own Swagger documents" %}
+```csharp
+using Microsoft.Extensions.Options;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace My.Custom.Swagger;
+
+public class MyConfigureSwaggerGenOptions : IConfigureOptions<SwaggerGenOptions>
+{
+    public void Configure(SwaggerGenOptions options)
+    {
+        options.SwaggerDoc(
+            "my-api-v1",
+            new OpenApiInfo
+            {
+                Title = "My API v1",
+                Version = "1.0",
+            });
+
+        options.SwaggerDoc(
+            "my-api-v2",
+            new OpenApiInfo
+            {
+                Title = "My API v2",
+                Version = "2.0",
+            });
+    }
+}
+
+public static class MyConfigureSwaggerGenUmbracoBuilderExtensions
+{
+    public static IUmbracoBuilder ConfigureMySwaggerGen(this IUmbracoBuilder builder)
+    {
+        // call this from ConfigureServices() in Startup, i.e.:
+        //     services.AddUmbraco(_env, _config)
+        //         ...
+        //         .ConfigureMySwaggerGen()
+        //         .Build();
+        builder.Services.ConfigureOptions<MyConfigureSwaggerGenOptions>();
+        return builder;
+    }
+}
+```
+{% endcode %}
+
+With these Swagger documents in place, you can now assign the different versions of your API controllers to their respective documents using the `MapToApi` annotation.
+
```

Comment:
For the following code snippets - would it make sense to put them into tabs instead?
That way you can quickly switch between them while avoiding having to scroll up and down to compare,

Reply:
Not really. We need to ensure that the readers include both versions of the API when testing. I reckon hiding one of them in tabs will increase the risk of that one being left out when testing, thus not achieving the desired result.

Reply:
Sure! Makes sense.